runtime/near-vm-runner/benches/bench.rs

```diff
@@ -52,68 +53,96 @@ fn assert_run_result((outcome, err): (Option<VMOutcome>, Option<VMError>), expec
-    bench.iter(move || {
-        let result = run(
-            vec![],
-            &code,
-            b"benchmark_storage_8b",
-            &mut external,
-            context.clone(),
-            &config,
-            &promise_results,
-        );
-        assert_run_result(result, 999 * 1000 / 2);
-    });
-}
+fn benchmark_fake_storage(c: &mut Criterion) {
+    let mut group = c.benchmark_group("benchmark_fake_storage");
+    let n_args = vec![10, 100, 1_000];
+    let size_args = vec![(8, 100), (1 << 10, 40), (10 << 10, 10)];
+    for (size, num_samples) in size_args {
+        group.sample_size(num_samples);
+        for n in n_args.clone() {
+            let benchmark_param_display = format!("n={}; size={}", n, size);
+            let expected_value = (n - 1) * n / 2;
 
-fn benchmark_fake_storage_10kib_1000(bench: &mut Bencher) {
-    let (mut external, context, config, promise_results, code) = setup(1000);
-    bench.iter(move || {
-        let result = run(
-            vec![],
-            &code,
-            b"benchmark_storage_10kib",
-            &mut external,
-            context.clone(),
-            &config,
-            &promise_results,
-        );
-        assert_run_result(result, 999 * 1000 / 2);
-    });
+            group.throughput(Throughput::Bytes(size * n));
+            let (mut external, context, config, promise_results, code) = setup(&[n, size]);
+            group.bench_function(BenchmarkId::new("wasm", benchmark_param_display.clone()), |b| {
+                b.iter(|| {
+                    let result = run(
+                        vec![],
+                        &code,
+                        b"benchmark_storage",
+                        &mut external,
+                        context.clone(),
+                        &config,
+                        &promise_results,
+                    );
+                    assert_run_result(result, expected_value as u64);
+                });
+            });
+        }
+    }
+    group.finish();
 }
 
-fn sum_n_1000000(bench: &mut Bencher) {
-    let (mut external, context, config, promise_results, code) = setup(1000000);
-    bench.iter(move || {
-        let result =
-            run(vec![], &code, b"sum_n", &mut external, context.clone(), &config, &promise_results);
-        assert_run_result(result, (1000000 - 1) * 1000000 / 2);
-    });
+fn sum_n(c: &mut Criterion) {
+    let mut group = c.benchmark_group("sum_n");
+    let args = vec![1, 100, 10_000, 1_000_000];
+    for n in args {
+        if n > 100_000 {
+            group.sample_size(20);
+        }
+        let (mut external, context, config, promise_results, code) = setup(&[n]);
+        let benchmark_param_display = format!("n={}", n);
+        let mut expected_value = 0u128;
+        for i in 1..n + 1 {
+            expected_value += (i * i) as u128;
+        }
+        expected_value /= n as u128;
+        group.bench_function(BenchmarkId::new("wasm", benchmark_param_display.clone()), |b| {
```

Comment:
benchmark id should include name of the function, e.g. "sum_n"?

Reply:
the group has the name